mantle/common/reg_file.py

```diff
@@ -0,0 +1,77 @@
+import magma as m
+from mantle.coreir import DefineRegister
+
+
+def _make_read_type(data_width, addr_width):
+    fields = dict(data=m.Out(m.Bits[data_width]), addr=m.In(m.Bits[addr_width]))
+    return m.Product.from_fields("ReadPortT", fields)
+
+
+def _make_write_type(data_width, addr_width):
+    fields = dict(data=m.In(m.Bits[data_width]), addr=m.In(m.Bits[addr_width]))
+    return m.Product.from_fields("WritePortT", fields)
+
+
+class RegFileBuilder(m.CircuitBuilder):
+    def __init__(self, name, height: int, width: int):
+        super().__init__(name)
+        self._data_width = width
+        self._height = height
+        self._addr_width = m.bitutils.clog2(height)
+        self._read_ports = []
+        self._write_ports = []
+        self._readT = _make_read_type(self._data_width, self._addr_width)
+        self._writeT = _make_write_type(self._data_width, self._addr_width)
+        clocks = m.ClockInterface(has_async_reset=True)
+        for name, typ in zip(clocks[::2], clocks[1::2]):
+            self._add_port(name, typ)
```

Comment:
Perhaps we can add an `__iter__` method for `IO`? Ideally we should deprecate the `ClockInterface` function and not use it in new code.

Reply:
Makes sense. Made a note of that here: https://github.com/phanrahan/magma/issues/596#issuecomment-603437721

For now I will use `ClockIO().decl()` instead of `ClockInterface`